crates/sqllib/src/casts.rs

```diff
@@ -2148,261 +2068,399 @@ pub fn negative(captures: &Captures) -> bool {
-            .parse()
-            .unwrap_or_else(|_| panic!("HOURS is not a number: {}", hcap));
+        let hourcap = captures.get(2).unwrap().as_str();
+        let hours = match hourcap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse HOURS: {}; {}",
+                    hourcap, e,
+                )))
+            }
+            Ok(hours) => hours,
+        };
         let mincap = captures.get(3).unwrap().as_str().trim();
-        let minutes = mincap
-            .parse()
-            .unwrap_or_else(|_| panic!("MINUTES is not a number: {}", mincap));
+        let minutes = match mincap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse MINUTES: {}; {}",
+                    mincap, e,
+                )))
+            }
+            Ok(minutes) => minutes,
+        };
         validate_minutes(minutes);
         let minutes = if negative {
             -hours * 60 - minutes
         } else {
             hours * 60 + minutes
         };
-        ShortInterval::new(minutes * 60 * 1000)
+        Ok(ShortInterval::new(minutes * 60 * 1000))
     } else {
-        panic!("Interval '{value}' does not have format 'hours:minutes'");
+        Err(SqlRuntimeError::from_string(format!(
+            "Interval '{value}' does not have format 'hours:minutes'",
+        )))
     }
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_ShortInterval_SECONDS_s(value: String) -> ShortInterval {
+pub fn cast_to_ShortInterval_SECONDS_s(value: String) -> SqlResult<ShortInterval> {
     if let Some(captures) = SECONDS.captures(&value) {
         let negative = negative(&captures);
         let seccap = captures.get(2).unwrap().as_str().trim();
-        let seconds: i64 = seccap
-            .parse()
-            .unwrap_or_else(|_| panic!("SECONDS is not a number: {}", seccap));
+        let seconds = match seccap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse SECONDS: {}; {}",
+                    seccap, e,
+                )))
+            }
+            Ok(seconds) => seconds,
+        };
         let ms = match captures.get(3) {
-            None => 0,
+            None => Ok(0i64),
             Some(_) => {
                 let mscap = captures.get(4).unwrap().as_str();
-                (mscap.to_owned() + "000000")[..3]
-                    .parse()
-                    .unwrap_or_else(|_| panic!("MILLISECONDS is not a number: {}", mscap))
+                (mscap.to_owned() + "000000")[..3].parse::<i64>()
+            }
+        };
+        let ms = match ms {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse MILLISECONDS: {}; {}",
+                    captures.get(4).unwrap().as_str(),
```

Comment:
what's 4, is it always there?

Reply:
it should be, because there are always 4 captures in the regular expression.